### PR TITLE
Build shared lib

### LINKED ISSUE #40: Allow dynamic library build
This is required for wrapping in Python, and some users may prefer to use a dynamic lib.

For our CMakeLists, should be easy. We'll probably have to figure out how to string-replace in the AMCL config python script, though (to make sure to include fPIC).

### PR DESCRIPTION
**Pull request by zanebeckwith**
Both static and shared libs can be built (if a shared lib is to be built, _all_ code will use fPIC).

Fixes #40

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -92,28 +113,98 @@ add_custom_target(toplevel_header
...
+################################################################################
+install(EXPORT ${CMAKE_PROJECT_NAME}Targets
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${CMAKE_PROJECT_NAME}
+        FILE        ${CMAKE_PROJECT_NAME}Config.cmake
+)
+
+################################################################################
+# Examples
+################################################################################
 if(ECDAA_BUILD_EXAMPLE_PROGRAMS)
```
Comment (drbild) [inline CMakeLists.txt]: `BUILD_EXAMPLES` seems to be the de facto standard.  Buildroot sets sets that to OFF by default.

File: CMakeLists.txt [hunk 2]
```diff
@@ -92,28 +113,98 @@ add_custom_target(toplevel_header
...
         set(ECDAA_PROGRAMS_DIR examples)
```
Comment (drbild) [inline CMakeLists.txt]: Does `ECDAA_PROGRAMS_DIR` need to be a variable, or can it just be a hard-coded `add_subdirectory(examples)`, like it is for `test`.

File: CMakeLists.txt [hunk 3]
```diff
@@ -92,28 +113,98 @@ add_custom_target(toplevel_header
...
         add_subdirectory(${ECDAA_PROGRAMS_DIR})
 endif(ECDAA_BUILD_EXAMPLE_PROGRAMS)
```
Comment (drbild) [inline CMakeLists.txt]: The examples and test flag checks aren't consistent.  The `BUILDING_TESTING` flag is checked inside `test/CMakeLists.txt`.  `BUILD_EXAMPLES` is checked here before the `add_subdirectory(examples)` call.

I don't have a reason to prefer one or the other, but we should probably standardize.

File: CMakeLists.txt (hunk 2 above)
Comment (zanebeckwith) [reply CMakeLists.txt]: No, no reason. I think that's left over from some early setup, but it's no longer needed. Removed.

File: CMakeLists.txt (hunk 3 above)
Comment (zanebeckwith) [reply CMakeLists.txt]: I've moved both to be in the top-level CMakeLists.txt. Good catch.

File: CMakeLists.txt (hunk 1 above)
Comment (zanebeckwith) [reply CMakeLists.txt]: Nice. Didn't realize that was a de facto standard.

**Pull request merged**